src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -174,6 +175,149 @@ Copyright (c) .NET Foundation. All rights reserved.
+    ============================================================
+    -->
+  <UsingTask TaskName="PrepareForReadyToRunCompilation" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="_PrepareForReadyToRunCompilation">
+
+    <PropertyGroup>
+      <_ReadyToRunOutputPath>$(IntermediateOutputPath)R2R</_ReadyToRunOutputPath>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(_ReadyToRunOutputPath)" />
+
+    <PrepareForReadyToRunCompilation TargetFramework="$(TargetFramework)"
+                                     RuntimePacks="@(ResolvedRuntimePack)"
+                                     RuntimeGraphPath="$(BundledRuntimeIdentifierGraphFile)"
+                                     KnownFrameworkReferences="@(KnownFrameworkReference)"
+                                     NETCoreSdkRuntimeIdentifier="$(NETCoreSdkRuntimeIdentifier)"
+                                     OutputPath="$(_ReadyToRunOutputPath)"
+                                     FilesToPublishAlways="@(_ResolvedFileToPublishAlways)"
+                                     FilesToPublishPreserveNewest="@(_ResolvedFileToPublishPreserveNewest)"
+                                     ReadyToRunExcludeList="@(ReadyToRunExclude)"
+                                     ReadyToRunEmitSymbols="$(ReadyToRunEmitSymbols)">
+
+      <Output TaskParameter="CrossgenTool" ItemName="_CrossgenTool" />
+
+      <Output TaskParameter="ReadyToRunCompileListPublishAlways" ItemName="_ReadyToRunCompileListPublishAlways" />
+      <Output TaskParameter="ReadyToRunSymbolsCompileListPublishAlways" ItemName="_ReadyToRunSymbolsCompileListPublishAlways" />
+
+      <Output TaskParameter="ReadyToRunCompileListPreserveNewest" ItemName="_ReadyToRunCompileListPreserveNewest" />
+      <Output TaskParameter="ReadyToRunSymbolsCompileListPreserveNewest" ItemName="_ReadyToRunSymbolsCompileListPreserveNewest" />
+
+      <Output TaskParameter="ReadyToRunFilesToPublishAlways" ItemName="_ReadyToRunFilesToPublishAlways" />
+      <Output TaskParameter="ReadyToRunFilesToPreserveNewest" ItemName="_ReadyToRunFilesToPreserveNewest" />
+
+    </PrepareForReadyToRunCompilation>
+
+  </Target>
+
+  <!--
+    ============================================================
+                                        _CreateR2RImagePreserveNewest
+
+    Compiles assemblies in the _ReadyToRunCompileListPreserveNewest list into ReadyToRun images
+    ============================================================
+    -->
+  <UsingTask TaskName="RunReadyToRunCompiler" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="_CreateR2RImagePreserveNewest"
+          Inputs="@(_ReadyToRunCompileListPreserveNewest)"
+          Outputs="%(_ReadyToRunCompileListPreserveNewest.OutputR2RImage)">
+
+    <RunReadyToRunCompiler CrossgenTool="@(_CrossgenTool)"
+                           CompilationEntry="@(_ReadyToRunCompileListPreserveNewest)"
+                           ContinueOnError="ErrorAndContinue" />
```

Comment:
From our experiment, you abandoned the part where we check the exit code and make sure that we abort the build steps that would happen after crossgen? Why?

Now we're just adding the failed items to the publish list and continuing? This seems wrong.

Reply:
I was thinking about this. The end result is that the build fails (although i think it might copy some files to the publish folder). I was just thinking that given that it shows an overall failing result, whatever is in the publish directory is not considered valid anyways, so it doesn't really matter what we add to the publish list... does this make sense?

Reply:
I'm not crazy about this because there will be an additional confounding error like "can't find foo.ni.dll" but only if it was a clean obj directory. Plus, we have to wonder about whatever other steps are not expecting the build to be in a error state but continuing.

Reply:
Fixed. Errors are getting detected now, and we stop building after the last crossgen step if there are any errors